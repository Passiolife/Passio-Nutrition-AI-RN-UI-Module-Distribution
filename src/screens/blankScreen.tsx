import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { WidgetsCard } from '../components';
import { ICONS } from '../assets';

export const BlankScreen = () => {
  return (
    <View style={styles.bodyContainer}>
      <Text>WIP</Text>
      <WidgetsCard
        widgetTitle="Water"
        leftIcon={ICONS.blueWaterDrop}
        rightIcon={ICONS.newAddPlus}
        value={45}
        unitValue="OZ"
      />
    </View>
  );
};

const styles = StyleSheet.create({
  bodyContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
});
