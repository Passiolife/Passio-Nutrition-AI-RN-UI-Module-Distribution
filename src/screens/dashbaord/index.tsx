export * from './DashboardScreen';
export * from './DashbaordScreen.Style';
export * from './utils';
