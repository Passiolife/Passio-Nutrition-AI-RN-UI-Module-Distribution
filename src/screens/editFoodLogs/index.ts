export * from './editFoodLogsScreen';
export * from './alerts';
export * from './utils';
