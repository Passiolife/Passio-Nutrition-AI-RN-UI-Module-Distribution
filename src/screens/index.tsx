export * from './nutritionProfile';
export * from './quick';
export * from './editFoodLogs';
export * from './scanning';
export * from '../models/FoodLog';
export * from '../models/FavoriteFoodItem';
export * from '../NutritionApp';
export * from './dashbaord';
export * from './meallogss';
export * from './foodSearch';
export * from './myFavoritess';
export * from './editIngredients';
export * from './recipeEditor';
export * from './onboardings';
export * from './myPlans';
export * from './home';
export * from './water';
export * from './weight';
export * from './nutritionInformation';
export * from './voiceLogging';
export * from './takePicture';
export * from './advisor';
export * from './imagePicker';
export * from './foodCreator';
export * from './myFoods';
export * from './barcode';
export * from './editRecipe';
