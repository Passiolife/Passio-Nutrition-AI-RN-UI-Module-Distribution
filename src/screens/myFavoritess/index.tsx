import MyFavoritesScreen from './MyFavoritesScreen';
export { MyFavoritesScreen };
export * from './MyFavoritesScreen';
export * from './views';
