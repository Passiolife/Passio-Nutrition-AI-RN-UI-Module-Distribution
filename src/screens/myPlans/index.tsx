export * from './MyPlanScreen';
export * from './mealPlan';
export * from './suggestions';
