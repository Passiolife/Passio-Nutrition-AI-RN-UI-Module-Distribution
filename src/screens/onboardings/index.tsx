import OnboardingScreen from './OnboardingScreen';
export { OnboardingScreen };
