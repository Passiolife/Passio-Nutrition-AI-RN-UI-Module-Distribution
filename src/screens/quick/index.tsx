export * from './views';
export * from './QuickScanningScreen';
