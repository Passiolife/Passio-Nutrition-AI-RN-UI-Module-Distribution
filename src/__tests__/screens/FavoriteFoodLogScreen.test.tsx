// @ts-nocheck
import React from 'react';
import {
  render,
  type RenderAPI,
  waitForElementToBeRemoved,
} from '@testing-library/react-native';
import { MockProviders } from '../provider/MockProviders';
import type { PassioIDAttributes } from '@passiolife/nutritionai-react-native-sdk-v3/src/sdk/v2';
import { MyFavoritesScreen } from '../../screens';

const mockedDispatch = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: mockedDispatch,
    }),
    useIsFocused: () => jest.fn(),
    useRoute: () => ({
      params: {
        logToDate: undefined,
        logToMeal: undefined,
      },
    }),
  };
});

jest.mock('@passiolife/nutritionai-react-native-sdk-v3', () => {
  const mockSDK = {
    getAttributesForPassioID(): Promise<PassioIDAttributes | null> {
      return require('../assets/json/mock_passio_sdk_attribute.json');
    },
  };
  const mockPassioIcon = require('../utils/MockPassioIcon').default;

  return {
    PassioSDK: mockSDK,
    PassioIconView: mockPassioIcon,
    IconSize: 'PX90',
  };
});

describe('FavoriteFoodLog Screen', () => {
  it('Renders correctly with no data', async () => {
    Date.now = jest.fn(() => new Date(2021, 9, 21, 14, 0, 0).valueOf());
    let tree: RenderAPI;
    tree = render(
      <MockProviders>
        <MyFavoritesScreen />
      </MockProviders>
    );

    await waitForElementToBeRemoved(() => tree.getByText('Please Wait....'));
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Renders correctly when favorites are present', async () => {
    Date.now = jest.fn(() => new Date(2021, 9, 21, 14, 0, 0).valueOf());
    let tree: RenderAPI;
    tree = render(
      <MockProviders>
        <MyFavoritesScreen />
      </MockProviders>
    );
    await waitForElementToBeRemoved(() => tree.getByText('Please Wait....'));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
