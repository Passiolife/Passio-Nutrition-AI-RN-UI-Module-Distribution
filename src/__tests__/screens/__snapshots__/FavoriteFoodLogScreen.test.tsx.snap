// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteFoodLog Screen Renders correctly when favorites are present 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    accessibilityIgnoresInvertColors={false}
    color="#123456"
    size="large"
  />
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#4A4A4A",
        "fontSize": 18,
        "fontWeight": "600",
        "paddingHorizontal": 16,
      }
    }
  >
    Please Wait....
  </Text>
</View>
`;

exports[`FavoriteFoodLog Screen Renders correctly with no data 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    accessibilityIgnoresInvertColors={false}
    color="#123456"
    size="large"
  />
  <Text
    style={
      {
        "alignSelf": "center",
        "color": "#4A4A4A",
        "fontSize": 18,
        "fontWeight": "600",
        "paddingHorizontal": 16,
      }
    }
  >
    Please Wait....
  </Text>
</View>
`;
